compiler/rustc_abi/src/lib.rs

```diff
@@ -274,6 +288,8 @@ pub struct TargetDataLayout {
     /// Note: This isn't in LLVM's data layout string, it is `short_enum`
     /// so the only valid spec for LLVM is c_int::BITS or 8
     pub c_enum_min_size: Integer,
+    /// Maximum size of #[repr(C)] enums (defaults to pointer size).
+    pub c_enum_max_size: Option<Integer>,
```

Comment:
kinda feels like we should just have a Range here, but on the other hand ranges usually have an unwieldy API so I'm not sure that's actually a good idea.

Reply:
...Mind, the enumeration constant detail is pure nonsense from Rust's POV, so it mostly serves to explain why C compilers have such a rough time figuring out how to handle this, since they didn't have the good sense to not extend the language with implementation-defined "sure, why not?"s. On some level they have to simultaneously pretend their enums are both `int`s and then some other type.

Reply:
Hmm. Thinking about it, I _guess_ they *can* hypothetically pick `intptr_t` as the type they're using, if that's a type that is held as distinct from any other integer type instead of a typedef? Hmm.